bin/_main

```diff
@@ -334,18 +334,35 @@ elif [ "${1}" == "run" ]; then
     benchmark_subproj_dir="${CRUCIBLE_HOME}"/subprojects/benchmarks/$benchmark
     rs_dir="${CRUCIBLE_HOME}"/subprojects/core/rickshaw
 
+    if [ ! -e "$benchmark_subproj_dir" ]; then
+        echo "Running benchmark $benchmark requires that subproject"
+        echo "located in "${CRUCIBLE_HOME}"/subprojects/bench/$benchmark"
+        echo "This directory could not be found.  Here are the benchmark"
+        echo "subproject directories:"
+        /bin/ls "${CRUCIBLE_HOME}"/subprojects/bench/$benchmark
+        exit 1
+    fi
+
     if [ "${use_mv_params}" == "1" ]; then
         if [ ! -e "${mv_params}" ]; then
             echo "The multi-value params file you specified with --mv-params (${mv_params}) does not exist!"
             exit 1
         else
+            echo "Generating --bench-params from --mv-params..."
+
             mv_params_run_dir=${base_run_dir}/config/mv-params.json
             cp ${mv_params} ${mv_params_run_dir}
             bench_params_run_dir=${base_run_dir}/config/bench-params.json
 
-            echo "Generating --bench-params from --mv-params..."
-            echo "$podman_run -i --name crucible-multiplex "${container_common_args[@]}" $CRUCIBLE_CONTAINER_IMAGE ${CRUCIBLE_HOME}/subprojects/core/multiplex/multiplex.py --input ${mv_params_run_dir} > ${bench_params_run_dir}"
-            $podman_run -i --name crucible-multiplex "${container_common_args[@]}" $CRUCIBLE_CONTAINER_IMAGE ${CRUCIBLE_HOME}/subprojects/core/multiplex/multiplex.py --input ${mv_params_run_dir} > ${bench_params_run_dir}
+            multiplex_cmd="${CRUCIBLE_HOME}/subprojects/core/multiplex/multiplex.py --input ${mv_params_run_dir}"
+            if [ -e ${benchmark_subproj_dir}/multiplex.json ]; then
+                multiplex_cmd+=" --requirements ${benchmark_subproj_dir}/multiplex.json"
```

Comment:
requirements file should be manually copied by the user to this dir ?

Reply:
no, the requirements file is part of the benchmark subproject -- ie. bench-fio has its own, bench-uperf has its own, etc.